### PR TITLE
[improve] [broker] PIP-355: Enhancing Broker-Level Metrics for Pulsar

### PR DESCRIPTION
**Pull request by hangc0276**
Fixes #xyz

Main Issue: #xyz

PIP: #22778 

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [x] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 1]
```diff
@@ -4342,5 +4352,4 @@ protected boolean isExceedMaximumDeliveryDelay(ByteBuf headersAndPayload) {
         }
         return false;
     }
-
```
Comment (shoothzj) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:4345]: unnecessary delete blank line

**Review by shoothzj — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java [hunk 2]
```diff
@@ -2559,6 +2562,8 @@ public CompletableFuture<? extends TopicStatsImpl> asyncGetStats(GetStatsOptions
         stats.waitingPublishers = getWaitingProducersCount();
         stats.bytesOutCounter = bytesOutFromRemovedSubscriptions.longValue();
         stats.msgOutCounter = msgOutFromRemovedSubscriptions.longValue();
+        stats.bytesOutInternalCounter = bytesOutFromRemovedSystemSubscriptions.longValue();
```
Comment (asafm) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:2565]: What about the values from an active (not removed) subscription? Aren't you missing them?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java [hunk 3]
```diff
@@ -157,10 +161,14 @@ public abstract class AbstractTopic implements Topic, TopicPolicyListener<TopicP
 
     protected final LongAdder msgOutFromRemovedSubscriptions = new LongAdder();
     protected final LongAdder bytesOutFromRemovedSubscriptions = new LongAdder();
+    protected final LongAdder bytesOutFromRemovedSystemSubscriptions = new LongAdder();
+    protected final LongAdder msgOutFromRemovedSystemSubscriptions = new LongAdder();
```
Comment (dao-jun) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:165]: The PIP not contains any metric like `pulsar_broker_out_msgs_total_internal`, please avoid this change

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java (hunk 2 above)
Comment (dao-jun) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java:2565]: No, seems not missed.
This line just add removed subscriptions stats, active system subscriptions' stats added: https://github.com/apache/pulsar/blob/352c316bd4c1f85b546ad70271feff78eae8f78a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java#L2595-L2596

**Review by dao-jun — CHANGES_REQUESTED** [review]
lgtm, just a few minor comments

**Review by dao-jun — APPROVED** [review]
LGTM, please fix the ci checks

**Pull request merged**